File: examples/typescript/council-debate.ts
import { ConsiliumClient } from "@consilium/sdk";
import type { DeliberationResult } from "@consilium/sdk";

async function runCouncilDebate(): Promise<void> {
  const client = new ConsiliumClient({
    apiKey: process.env.CONSILIUM_API_KEY ?? "your-api-key-here",
    apiUrl: process.env.CONSILIUM_API_URL ?? "http://localhost:3000/api",
  });

  const architectureTopic = [
    "Should we adopt a CQRS/Event Sourcing architecture for our fintech",
    "transaction processing system that handles 2M daily transactions,",
    "or stick with traditional CRUD with optimistic locking? Consider:",
    "audit requirements, replay capability, eventual consistency tradeoffs,",
    "team expertise ramp-up, and operational complexity.",
  ].join(" ");

  const panelistModels = [
    "gpt-4o",
    "claude-sonnet-4-20250514",
    "gemini-2.0-flash",
  ];

  const healthStatus = await client.healthCheck();
  if (healthStatus.status !== "ok") {
    console.error(`API unhealthy: ${healthStatus.status}`);
    process.exit(1);
  }

  const costEstimate = await client.estimateCost({
    topic: architectureTopic,
    mode: "council",
  });
  console.log(`Estimated cost: $${costEstimate.estimatedCost.toFixed(4)}\n`);

  const result: DeliberationResult = await client.deliberate({
    topic: architectureTopic,
    mode: "council",
    models: panelistModels,
    maxRounds: 3,
  });

  console.log("=== Council Deliberation Result ===\n");
  console.log(`Golden Prompt:\n${result.goldenPrompt}\n`);
  console.log(`Dissent Report:\n${result.dissentReport}\n`);
  console.log(`Total Cost: $${result.cost.toFixed(4)}\n`);

  console.log("Confidence Scores:");
  for (const [model, score] of Object.entries(result.confidenceScores)) {
    console.log(`  ${model}: ${(score * 100).toFixed(1)}%`);
  }

  console.log(`\nAudit Trail: ${result.auditTrail.length} entries`);
}

runCouncilDebate().catch(console.error);

File: examples/python/council_debate.py
import os
import sys

from consilium import ConsiliumClient, DeliberationMode, ConsiliumError


def run_architecture_debate():
    client = ConsiliumClient(
        api_key=os.environ.get("CONSILIUM_API_KEY", "your-api-key-here"),
        api_url=os.environ.get("CONSILIUM_API_URL", "http://localhost:4000/api/v1"),
    )

    architecture_topic = (
        "Should we migrate our monolithic Django application to microservices "
        "using Kubernetes, or refactor into a modular monolith with clear domain "
        "boundaries? The app serves 50k daily active users with 99.9% uptime SLA."
    )

    panelist_models = [
        "gpt-5.4",
        "claude-sonnet-4-6",
        "gemini-3-flash-preview",
    ]

    try:
        result = client.deliberate(
            topic=architecture_topic,
            models=panelist_models,
            mode=DeliberationMode.COUNCIL,
            max_rounds=3,
        )
    except ConsiliumError as exc:
        print(f"Deliberation failed: {exc}")
        sys.exit(1)

    print("=== Council Deliberation Result ===\n")
    print(f"Golden Prompt:\n{result.golden_prompt}\n")
    print(f"Dissent Report:\n{result.dissent_report}\n")
    print(f"Total Cost: ${result.cost:.4f}\n")

    print("Confidence Scores:")
    for model_name, score in result.confidence_scores.items():
        print(f"  {model_name}: {score:.1%}")

    print(f"\nAudit Trail: {len(result.audit_trail)} entries")


if __name__ == "__main__":
    run_architecture_debate()
